File: javascript/easy/2520.countDigits.js
/**
 * @param {number} num
 * @return {number}
 */
var countDigits = function (num) {
    const s = String(num).split("")
    let counter = 0

    for (let i = 0; i < s.length; i++) {
        if (num % s[i] == 0) {
            counter++
        }
    }

    return counter
}

console.log(countDigits(121)) // 2
// Output: 2
// Explanation: 121 is divisible by 1, but not 2. Since 1 occurs twice as a digit, we return 2.

console.log(countDigits(1248)) // 4

File: python/easy/2520.countDigits.py
class Solution:
    def countDigits(self, num: int) -> int:
        counter: int = 0

        for digit in str(num):
            if num % int(digit) == 0:
                counter += 1

        return counter


if __name__ == "__main__":
    solution = Solution()
    # print(solution.countDigits(7))  # 1
    print(solution.countDigits(121))  # 2
